### PR TITLE
Refactor bar and improve documentation

### PR DESCRIPTION
**Pull request by owennw**
Small tweak to the bar class ahead of implementing variable widths, and update to the documentation to showcase the existing functionality.

### CODE REVIEW

File: site/src/components/series/bar.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ component: series/bar.js
 tags:
 namespace: series
 
-example-code: |
+example-code-1: |
```
Comment (jdunkerley) [inline site/src/components/series/bar.md]: no need to change from example-code and example-fixture

**Comment by jdunkerley** [comment]
looks good, but would switch back `example-code-1` to `example-code` and keep the `example-fixture` as no need to change and forthcoming playground integration based on this at moment.

**Comment by jdunkerley** [comment]
:+1:

**Comment by ColinEberhardt** [comment]
I like the extra documentation, it's a good idea that we demonstrate the capabilities of this library more clearly. 

However, I'm not so keen on the refactor to the bar series component. I don't think it makes the code any clearer or more efficient, it's just different!

Also, it is a little inconsistent in that it adds an `isVertical` function but not a corresponding `isHorizontal`.

I'd be happy to merge the documentation update, but keep the code as-is. Sorry!

**Comment by owennw** [comment]
Adding an `isHorizontal` is trivial, but the main cause for the refactor is to remove duplication, and only specify a piece of logic once. If we wanted to change the string that specified the orientation we now only have to do that in one place, not several. (The same is currently true for horizontal, which is why I left it there).

I think in general not repeating yourself and expressing a concept once is both clearer, and from a developer's point-of-view, more efficient.

**Comment by ColinEberhardt** [comment]
Unfortunately code quality is subjective and what constitutes clearer code
depends on opinion. While I do agree with the DRY principles in their most
general sense, my feeling is that they lose value when applied to such small
units of code.
Furthermore, the features of a particular language have an influence on the
conventions that developers adopt. You'll find that JavaScript developers are
much more comfortable with repeated string literals than a C# or Java developer
might be.
For guidance take a look at the d3 source code.

On Fri, Oct 30, 2015 at 1:45 p.m., ScottLogic/d3fc
<reply@reply.github.com>
wrote:
Adding an isHorizontal is trivial, but the main cause for the refactor is to remove duplication, and
only specify a piece of logic once. If we wanted to change the string that
specified the orientation we now only have to do that in one place, not several.
(The same is currently true for horizontal, which is why I left it there).

I think in general not repeating yourself and expressing a concept once is both
clearer, and from a developer's point-of-view, more efficient.

—
Reply to this email directly or view it on GitHub
[https://github.com/ScottLogic/d3fc/pull/713#issuecomment-152529881] .[https://github.com/notifications/beacon/ABDBfmTX_sWQkaD6eWN2DmXTlyz4m9P3ks5pA2vhgaJpZM4GY5d8.gif]

**Comment by owennw** [comment]
I agree that aspects of code quality can be subjective, but I haven't seen a reason for why this particular refactor doesn't have, or loses, value.

>If we wanted to change the string that specified the orientation we now only have to do that in one place, not several.

While JS devs might be more comfortable with repeated logic, I don't see why that is sufficient reason to ignore the benefit pointed out above.

**Comment by jdunkerley** [comment]
Could we do something taking this a bit further:

``` JavaScript
orientations = {
    vertical {
        containerTranslation: function(d, i) { return 'translate(' + base.x1(d, i) + ', ' + base.y0(d, i) + ')'; }
        barHeight: function(d, i) { return base.y1(d, i) - base.y0(d, i); },
        valueAxisDimension ...
    }, 
    horizontal {
        ...
    }
}

var bar = function(selection) {
    selection.each(function(data, index) {

        if (orient !== 'vertical' && orient !== 'horizontal') {
            throw new Error('The bar series does not support an orientation of ' + orient);
        }
```

**Pull request closed without merging**